products/mlengine/api.yaml

```diff
@@ -73,3 +91,148 @@ objects:
+        output: true
+      - !ruby/object:Api::Type::Time
+        name: 'lastUseTime'
+        description: The time the version was last used for prediction.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'runtimeVersion'
+        description: The AI Platform runtime version to use for this deployment
+      - !ruby/object:Api::Type::Enum
+        name: 'machineType'
+        description: The type of machine on which to serve the model. Currently only applies to online prediction service.
+        values:
+          - mls1-c1-m2
+          - mls1-c4-m2
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: The state of a version
+        values:
+          - UNKNOWN
+          - READY
+          - CREATING
+          - FAILED
+          - DELETING
+          - UPDATING
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'errorMessage'
+        description: The details of a failure or cancellation
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: 'packageUris'
+        description: Cloud Storage paths (gs://…) of packages for custom prediction routines or scikit-learn pipelines with custom code.
+        output: true
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: One or more labels that you can add, to organize your model versions
+        # no etag since it's input: true
+      - !ruby/object:Api::Type::Enum
+        name: 'framework'
+        description: The machine learning framework AI Platform uses to train this version of the model
+        values:
+          - FRAMEWORK_UNSPECIFIED
+          - TENSORFLOW
+          - SCIKIT_LEARN
+          - XGBOOST
+      - !ruby/object:Api::Type::String
+        name: 'pythonVersion'
+        description: |
+          The version of Python used in prediction. If not set, the default
+          version is '2.7'. Python '3.5' is available when runtimeVersion is
+          set to '1.4' and above. Python '2.7' works with all supported runtime
+          versions.
+      - !ruby/object:Api::Type::String
+        name: 'serviceAccount'
+        description: Specifies the service account for resource access control.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'autoScaling'
+        description: |
+          Automatically scale the number of nodes used to serve the model in
+          response to increases and decreases in traffic. Care should be taken
+          to ramp up traffic according to the model's ability to scale or you
+          will start seeing increases in latency and 429 response codes.
+        conflicts: ['manualScaling']
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'minNodes'
+            description: The minimum number of nodes to allocate for this mode
+      - !ruby/object:Api::Type::NestedObject
```

Comment:
Is it possible to flatten the scaling-related nested objects' single member fields into top-level params?

Reply:
Ansible doesn't have this feature currently (I believe TF does).